build.gradle.kts

```diff
@@ -0,0 +1,372 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import com.adarshr.gradle.testlogger.TestLoggerExtension
+import com.adarshr.gradle.testlogger.TestLoggerPlugin
+import org.jetbrains.intellij.IntelliJPluginExtension
+import org.jetbrains.intellij.tasks.PrepareSandboxTask
+import org.jetbrains.intellij.tasks.PublishTask
+import org.jetbrains.intellij.tasks.RunIdeTask
+import org.jetbrains.intellij.tasks.RunIdeForUiTestTask
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+import software.aws.toolkits.gradle.IdeVersions
+import software.aws.toolkits.gradle.ProductCode
+import software.aws.toolkits.gradle.changelog.tasks.GenerateGithubChangeLog
+import software.aws.toolkits.gradle.findFolders
+import software.aws.toolkits.gradle.getOrCreate
+import software.aws.toolkits.gradle.resources.ValidateMessages
+
+buildscript {
+    repositories {
+        maven("https://plugins.gradle.org/m2/")
+        mavenCentral()
+        jcenter()
+    }
+    val kotlinVersion: String by project
+    val ideaPluginVersion: String by project
+    dependencies {
+        classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion")
+        classpath("gradle.plugin.org.jetbrains.intellij.plugins:gradle-intellij-plugin:$ideaPluginVersion")
+        classpath("com.adarshr:gradle-test-logger-plugin:1.7.0")
+    }
+}
+
+val ideVersions = IdeVersions(project)
+val ideVersion = ideVersions.resolveShortenedIdeProfileName()
+val toolkitVersion: String by project
+val kotlinVersion: String by project
+val mockitoVersion: String by project
+val mockitoKotlinVersion: String by project
+val assertjVersion: String by project
+val junitVersion: String by project
+val remoteRobotPort: String by project
+val ktlintVersion: String by project
+val remoteRobotVersion: String by project
+val ideaPluginVersion: String by project
+// publishing
+val publishToken: String by project
+val publishChannel: String by project
+
+plugins {
+    java
+    id("de.undercouch.download") version "4.1.1" apply false
+}
+
+group = "software.aws.toolkits"
+// please check changelog generation logic if this format is changed
+version = "$toolkitVersion-$ideVersion".toString()
+
+repositories {
+    maven("https://www.jetbrains.com/intellij-repository/snapshots/")
+}
+
+allprojects {
+    repositories {
+        mavenLocal()
+        mavenCentral()
+        jcenter()
+    }
+
+    apply(plugin = "com.adarshr.test-logger")
+    apply(plugin = "java")
+    apply(plugin = "jacoco")
+
+    java.sourceCompatibility = JavaVersion.VERSION_1_8
+    java.targetCompatibility = JavaVersion.VERSION_1_8
+
+    tasks.withType(JavaExec::class.java) {
+        systemProperty("aws.toolkits.enableTelemetry", false)
+    }
+
+    tasks.withType(RunIdeTask::class.java) {
+        val ijExt = project.extensions.getByName("intellij") as IntelliJPluginExtension
```

Comment:
You can do `getByType` eg: `project.extensions.getByType<IntelliJPluginExtension>()` saves you having to do the cast and you don't need to figure out the name.